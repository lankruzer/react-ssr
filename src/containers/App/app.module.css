body:global {
    margin: 0;
    padding: 0;
    font-size: 18px;
    line-height: 1.26;
    font-family: sans-serif;
}

.appContainer {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
}