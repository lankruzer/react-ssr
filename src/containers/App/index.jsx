import React, { useEffect } from 'react';
import Footer from '../../components/Footer';
import styles from './app.module.css';
import FilmsContextHOC from '../../HOCs/FilmsContextHOC';
import ErrorBoundary from '../ErrorBoundary';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import PageFilmList from './../PageFilmList';
import PageFilmItem from './../PageFilmItem';

export default () => {
    const PageNotFound = () => {
        return (
            <div>
                <h1>Page not found</h1>
            </div>
        );
    };

    return (
        <div className={styles.appContainer}>
            <ErrorBoundary>
                <Router>
                    <FilmsContextHOC>
                        <Switch>
                            <Route exact path="/" component={PageFilmList} />
                            <Route
                                exact
                                path="/search/:searchQuery"
                                component={PageFilmList}
                            />
                            <Route
                                exact
                                path="/film/:filmId"
                                component={PageFilmItem}
                            />
                            <Route component={PageNotFound} />
                        </Switch>
                    </FilmsContextHOC>
                    <Footer />
                </Router>
            </ErrorBoundary>
        </div>
    );
};
