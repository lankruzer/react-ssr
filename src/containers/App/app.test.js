import React from 'react';
import {mount} from 'enzyme';
import App from './index';

describe('App render', () => {
    it('should render App', () => {
        const component = mount(<App/>);
        expect(component).toMatchSnapshot();

        component.unmount();
    });
});