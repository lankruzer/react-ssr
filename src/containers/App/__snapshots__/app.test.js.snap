// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should render App 1`] = `
<_default>
  <div>
    <main>
      <div>
        Hi! My black brothers!!!
      </div>
    </main>
    <_default>
      <footer>
        <a
          href="/"
        >
          netflix
        </a>
      </footer>
    </_default>
  </div>
</_default>
`;
